src/include/txn.i

```diff
@@ -193,6 +193,172 @@ __wt_timestamp_subone(wt_timestamp_t *ts)
 
 #endif /* HAVE_TIMESTAMPS */
 
+/*
+ * __txn_op_modify_recno --
+ *      Modify the recno in the latest txn op.
+ */
+static inline void
+__txn_op_modify_recno(WT_SESSION_IMPL *session, uint64_t recno)
+{
+	WT_TXN *txn;
+	WT_TXN_OP *op;
+
+	txn = &session->txn;
+
+	WT_ASSERT(session, txn->mod_count > 0 && recno != WT_RECNO_OOB);
+	op = txn->mod + txn->mod_count - 1;
+
+	WT_ASSERT(session, op->type == WT_TXN_OP_BASIC_COL ||
+	    op->type == WT_TXN_OP_INMEM_COL);
+
+	op->u.op_col.recno = recno;
+}
+
+/*
+ * __txn_op_resolve --
+ *      Resolve a transaction operation indirect references.
+ */
+static inline int
+__txn_op_resolve(WT_SESSION_IMPL *session, WT_TXN_OP *op, bool commit)
+{
+	WT_CURSOR *cursor;
+	WT_DECL_RET;
+	WT_TXN *txn;
+	WT_UPDATE *upd;
+	const char *open_cursor_cfg[] = {
+	    WT_CONFIG_BASE(session, WT_SESSION_open_cursor), NULL };
+
+	txn = &session->txn;
+
+	switch (op->type) {
+	case WT_TXN_OP_NONE:
+		break;
+	case WT_TXN_OP_BASIC_COL:
+	case WT_TXN_OP_BASIC_ROW:
+	case WT_TXN_OP_INMEM_COL:
+	case WT_TXN_OP_INMEM_ROW:
+		cursor = NULL;
+		WT_ERR(__wt_open_cursor(session, op->btree->dhandle->name,
+		    NULL, open_cursor_cfg, &cursor));
+		if (op->type == WT_TXN_OP_BASIC_ROW ||
```

Comment:
Suggestion for restructuring are most welcome. I don't like the if condition on "op->type" again with in a switch which is based on "op->type".